### PR TITLE
change back to original matplotlib backend

### PR DESCRIPTION
**Pull request by sshen8**
When using the UMI-tools API only, I noticed it switches my Matplotlib backend to Agg, which messes with the plots I generate downstream. This PR changes the backend only when generating plots and then switches back to the user's original setting

### CODE REVIEW

**Comment by TomSmithCGAT** [comment]
Thank you @sshen8. This looks like how I should have implemented it in the first place!

@IanSudbery - I'll merge this once tests pass.

File: umi_tools/whitelist_methods.py [hunk 1]
```diff
@@ -100,6 +100,9 @@ def getKneeEstimateDensity(cell_barcode_counts,
 
     if plotfile_prefix:
 
+        # require to run on systems with no X11
```
Comment (IanSudbery) [inline umi_tools/whitelist_methods.py:103]: @sshen8 How would you feel about moving lines 11 and 12 to just here (line 102). This way, if the user changes any params after importing umi_tools, then those changes will get captured.

**Comment by sshen8** [comment]
Yeah that's a good idea! I've just made the change, thanks for thinking of it

**Comment by IanSudbery** [comment]
@TomSmithCGAT Happy once the tests complete.

**Pull request merged**